test/CMakeLists.txt

```diff
@@ -1,24 +1,21 @@
 include(GoogleTest)
 include(CTest)
 
-add_executable(loggerTest
-    AddTest.cpp
+add_executable(LogEntryTest
+    LogEntryTest.cpp
 )
+target_link_libraries(LogEntryTest PRIVATE logentry gtest_main)
+add_test(NAME LogEntryTest COMMAND LogEntryTest)
 
-target_include_directories(loggerTest PUBLIC
-    "${gtest_SOURCE_DIR}/src"
-    "${CMAKE_SOURCE_DIR}/src"
+add_executable(LogParserTest
+    LogParserTest.cpp
 )
-
-target_link_libraries(loggerTest PRIVATE
-    calculator
-    gtest_main
-    gmock_main
-)
-
-gtest_discover_tests(loggerTest XML_OUTPUT_DIR ${CMAKE_BINARY_DIR}/test/loggerTest)
+target_link_libraries(LogParserTest PRIVATE logparser logentry gtest_main)
```

Comment:
if you need to link both logentry and logparser it means it is not an unit test :P

In this case you should define abstract interface for the logentry and crate mock to use it in tests :)

... it can be done in the separate PR tho